packages/fiber/src/core/hooks.tsx

```diff
@@ -51,6 +52,22 @@ export function useFrame(callback: RenderCallback, renderPriority: number = 0):
   return null
 }
 
+/**
+ * Executes a callback in a given update stage.
+ * Uses the stage instance to indetify which stage to target in the lifecycle.
+ */
+export function useUpdate(callback: UpdateCallback, stage: StageTypes = Stages.Update) {
+  const store = useStore()
+  const stages = store.getState().internal.stages
+  const match = stages.find((s) => s === stage)
+  // Memoize ref
+  const ref = useMutableCallback(callback)
+  // Throw an error if a stage does not exist
+  if (!match) throw `A '${stage}' stage does not exist.`
```

Comment:
Since `stage` is now the actual object, I don't think logging it this way still makes sense. In fact, the above line where you get the `match` now feels a little obsolete, since it just returns the same object. Maybe this can be changed to `includes(stage)` if the intention is to do a sanity check to see if the stage is actually known to the Canvas' lifecycle?